File: frontend/src/api/performance.ts
import { apiFetch } from "@/api/httpClient";
import type { PerformanceSummary } from "@/features/performance/types";

export async function getPerformanceSummary(): Promise<PerformanceSummary> {
  return apiFetch<PerformanceSummary>("/api/v1/performance/summary");
}

File: backend/app/api/routes/performance.py
from __future__ import annotations

from typing import Annotated, Any

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.deps import get_workspace_context
from app.core.tenant import TenantContext
from app.infrastructure.database import get_db_session
from app.services import performance_service

router = APIRouter(prefix="/api/v1/performance", tags=["performance"])


@router.get("/summary")
async def performance_summary(
    session: Annotated[AsyncSession, Depends(get_db_session)],
    tenant: Annotated[TenantContext, Depends(get_workspace_context)],
) -> dict[str, Any]:
    return await performance_service.workspace_performance_summary(session, tenant)
